### PR TITLE
BSTOR-42 - update existing items quantity when adding new items

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

File: client/src/test/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/CartItemPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @Before
+    public void setUp() {
+        currentShoppingCartItem = new ShoppingCartItem(productDto, initialQuantity);
+
+        presenter = cartItemFactory.create(currentShoppingCartItem);
+    }
+
+    @Test
+    public void onShoppingCartQuantityUpdated_updatesViewWithNewQuantity_whenCartItemIsSameAsCurrentOne() {
+        int newQuantity = 414;
```
Comment (olafleur) [inline client/src/test/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/CartItemPresenterTest.java]: Maybe extract in a constant or at least capitalize the name and add a const?

**Comment by olafleur** [comment]
LGTM

File: client/src/main/java/com/arcbees/beeshop/client/application/CurrentOrderImpl.java [hunk 2]
```diff
@@ -39,10 +42,21 @@
     }
 
     @Override
-    public void addItem(ShoppingCartItem item) {
-        items.add(item);
-
-        ShoppingCartChangedEvent.fire(item, this);
+    public void addItem(final ShoppingCartItem item) {
+        ShoppingCartItem existingItemOfSameType = Iterables.find(items, new Predicate<ShoppingCartItem>() {
```
Comment (meriouma) [inline client/src/main/java/com/arcbees/beeshop/client/application/CurrentOrderImpl.java]: change to `Iterables.tryFind(items, predicate).orNull()`, it's more verbose

**Comment by meriouma** [comment]
LGTM

File: client/src/test/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/CartItemPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Inject
+    private CartItemFactory cartItemFactory;
+
+    private CartItemPresenter presenter;
+    private ProductDto productDto;
+    private int initialQuantity;
+    private ShoppingCartItem currentShoppingCartItem;
+    @Inject
+    private CartItemPresenter.MyView view;
```
Comment (jclariviere) [inline client/src/test/java/com/arcbees/beeshop/client/application/widget/sidepanel/cart/CartItemPresenterTest.java:51]: Put with other `@Inject` fields

**Comment by jclariviere** [comment]
LGTM

**Pull request merged**